apps/SageAccountWeb/src/App.tsx

```diff
@@ -32,44 +30,34 @@ import TwoFactorAuthBackupCodesPage from './components/TwoFactorAuth/TwoFactorAu
 import TwoFactorAuthEnrollmentPage from './components/TwoFactorAuth/TwoFactorAuthEnrollmentPage'
 import { WebhookManagementPage } from './components/WebhooksManagementPage'
 import { RESET_2FA_ROUTE } from './Constants'
+import useMaybeRedirectToSignTermsOfService from './hooks/useMaybeRedirectToSignTermsOfService'
 import LoginPage from './LoginPage'
 import { getSearchParam } from './URLUtils'
 import './App.scss'
 
 function LoggedInRedirector() {
   const { accessToken } = useSynapseContext()
   const appContext = useAppContext()
-  const { data: tosStatus } = SynapseQueries.useTermsOfServiceStatus(
-    accessToken,
-    {
-      enabled: !!accessToken,
-    },
-  )
 
   const isCodeSearchParam = getSearchParam('code') !== undefined
   const isProviderSearchParam = getSearchParam('provider') !== undefined
   const isInSSOFlow = isCodeSearchParam && isProviderSearchParam
-  const history = useHistory()
+
+  const { mayRedirect: mayRedirectToSignToS } =
+    useMaybeRedirectToSignTermsOfService()
```

Comment:
Extracted the ToS redirect into a shared hook, which returns a `mayRedirect` flag that we can use to decide if we should run the redirect at the root route